### PR TITLE
PageStorage: Fix the WAL snapshot dumped into file may contains invalid "being_ref_count"

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/4957, close https://github.com/pingcap/tiflash/issues/4986

Problem Summary:

* issue 4975:
  * We can not prevent `being_ref_count` to be modify unless we stop the write threads while dumping snapshot of WAL log files
* issue 4986:
  * The encryption info of WAL log files are not removed after snapshot dumped

### What is changed and how it works?

* Create a reader for WAL log files and load the edit into another temporary PageDirectory instance, and dump it as a snapshot edit to the snapshot log file
* Remove the encryption info of WAL log files after snapshot dumped

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [x] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/WALStore.cpp [hunk 1]
```diff
@@ -212,25 +219,11 @@ bool WALStore::saveSnapshot(FilesSnapshot && files_snap, PageEntriesEdit && dire
...
-            Poco::File archive_dir(archive_path);
-            if (!archive_dir.exists())
-                archive_dir.createDirectory();
-            auto dest = archive_path.toString() + "/" + filename.filename(LogFileStage::Normal);
-            f.moveTo(dest);
-            LOG_FMT_INFO(logger, "archive {} to {}", filename.fullname(LogFileStage::Normal), dest);
-#endif
-        }
+        const auto log_fullname = filename.fullname(LogFileStage::Normal);
+        provider->deleteRegularFile(log_fullname, EncryptionPath(log_fullname, ""));
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/WALStore.cpp:226]: Fix #4986

[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Storages/Page/V3/WALStore.cpp:226]: `provider->deleteRegularFile` will check whether the file exist or not. And we don't need to do anything if the file is already removed
https://github.com/pingcap/tiflash/blob/64a747a179d736475a66e02d926efbab5a133c32/dbms/src/Encryption/FileProvider.cpp#L136-L139

**Comment by JaySon-Huang** [comment]
Addressed, PTAL again @jiaqizho

**Review by jiaqizho — APPROVED** [review]
LGTM

**Review by lidezhu — APPROVED** [review]

File: dbms/src/Storages/Page/PageStorage.h [hunk 2]
```diff
@@ -140,7 +140,7 @@ class PageStorage : private boost::noncopyable
         SettingUInt64 blob_block_alignment_bytes = 0;
 
         SettingUInt64 wal_roll_size = PAGE_META_ROLL_SIZE;
-        SettingUInt64 wal_recover_mode = 0;
+        SettingUInt64 wal_recover_mode = 0x00; // WALRecoveryMode::TolerateCorruptedTailRecords
```
Comment (flowbehappy) [inline dbms/src/Storages/Page/PageStorage.h:143]: Why not simply 
```suggestion
        SettingUInt64 wal_recover_mode = WALRecoveryMode::TolerateCorruptedTailRecords;
```

**Pull request merged**